scripts/build_test.py

```diff
@@ -679,14 +676,16 @@ def test_generate_app_yaml_with_deploy_mode(self):
         )
 
         app_dev_yaml_temp_file = tempfile.NamedTemporaryFile()
-        app_dev_yaml_temp_file.name = mock_dev_yaml_filepath
+        setattr(
+            app_dev_yaml_temp_file, app_dev_yaml_temp_file.name,
+            mock_dev_yaml_filepath)
```

Comment:
Why is this needed?

Reply:
Previously, I was getting this error.
### scripts/build_test.py:684: error: Property "name" defined in "IO" is read-only  [misc]
and from these error logs, it seems like we are trying to set an attribute of a read-only property and for that, I think we need to use `setattr` instead of an assignment operator. please correct me if I am wrong.